### PR TITLE
Adding codecov + lgtm to CI

### PR DESCRIPTION
**Pull request by tlfobe**
Using this branch to add lgtm and codecov to the CI workflow

I'll also add some cool badges!

### CODE REVIEW

**Comment by tlfobe** [comment]
I believe this is finished. I also changed a variable in `physical_validation/util/gromacs_interface.py` to get pass a lint test error

File: .github/workflows/continous_integration.yaml [hunk 1]
```diff
@@ -74,9 +74,4 @@ jobs:
           pytest -v --cov=physical_validation physical_validation/tests/
```
Comment (SimonBoothroyd) [inline .github/workflows/continous_integration.yaml:74]:
```suggestion
          pytest -v --cov=physical_validation --cov-report=xml physical_validation/tests/
```
For some reason I have found on my other repos that I needed to explicitly ask for an xml report be generated in order for codecov to work correctly

File: .github/workflows/continous_integration.yaml [hunk 2]
```diff
@@ -74,9 +74,4 @@ jobs:
...
 
       - name: Code coverage
-        run: |
-
-          . devtools/scripts/initialize_conda.sh
-          conda activate test
-
-          codecov
+        uses: codecov/codecov-action@v1
```
Comment (SimonBoothroyd) [inline .github/workflows/continous_integration.yaml:77]:
```suggestion
        uses: codecov/codecov-action@v1
        with:
          file: ./coverage.xml
```
Same as the above comment - I've found in the past I needed to explicitly give codecov the expected file path for it to upload correctly.

Comment (tlfobe) [reply .github/workflows/continous_integration.yaml:77]: Good call on explicitly telling pytest to write an xml report file. I though pytest was writing it in an obscure location, but it turns out it wasn't writing it at all!

Comment (tlfobe) [reply .github/workflows/continous_integration.yaml:77]: Added those changes and now the this commits report popped up on codecov. (see the codecov-commenter below)

**Pull request merged**